File: API/Core/Entities/Dres.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class Dres : BaseEntity
    {
        [Key]
        public Guid DresId { get; set; }

      
        [StringLength(40)]
        public string ImeIgraca { get; set; }

        [Required]
        [StringLength(30)]
        public string Tim { get; set; }

        [Required]
        [StringLength(10)]
        public string Sezona { get; set; }

        [Required]
        [StringLength(20)]
        public string Brend { get; set; }
        public double Cena { get; set; }

        [Required]
        
        public string SlikaUrl { get; set; }

        
        public bool Obrisan { get; set; }

        [Required]
        [StringLength(10)]
        public string Tip { get; set; }

        [Required]
        [StringLength(30)]
        public string Zemlja { get; set; }

        [Required]
        [StringLength(30)]
        public string Takmicenje { get; set; }

        [Required]
        [StringLength(10)]
        public string Status { get; set; }

        [Required]
        [StringLength(500)]
        public string TimUrl { get; set; }
        public Admin Admin { get; set; }
        public Guid AdminId { get; set; }

        [JsonIgnore]
        public ICollection<VelicinaDresa> VelicinaDresas { get; set; }


    }
}

File: Frontend/ClientApp/src/app/shared/models/dres.ts
import { Admin } from "./admin";

export class Dres {
       dresId: string
        imeIgraca: string
        tim: string
         sezona: string
         brend: string
         cena: number
         slikaUrl: string
         tip: string
         obrisan: boolean
         zemlja: string
         takmicenje: string
         status: string
         timUrl: string
         admin: Admin
         adminId: string
        
         
         constructor(dresId: string, imeIgraca: string, tim:string, sezona: string, brend: string, cena: number, slikaUrl: string, tip: string,
            obrisan: boolean, zemlja: string, takmicenje: string, status: string, timUrl: string, admin: Admin, adminId: string){
                this.dresId=dresId
                this.imeIgraca=imeIgraca;
                this.tim=tim;
                this.sezona=sezona;
                this.brend=brend;
                this.cena=cena;
                this.slikaUrl=slikaUrl;
                this.tip=tip
                this.obrisan=obrisan
                this.zemlja=zemlja;
                this.takmicenje=takmicenje;
                this.status=status;
                this.timUrl=timUrl;
                this.admin=admin;
                this.adminId=adminId
            }
     
          
    

}


export class DresCreation {
   
     imeIgraca: string
     tim: string
      sezona: string
      brend: string
      cena: number
      slikaUrl: string
      tip: string
      obrisan: boolean
      zemlja: string
      takmicenje: string
      status: string
      timUrl: string
      adminId: string
     
      
      constructor( imeIgraca: string, tim:string, sezona: string, brend: string, cena: number, slikaUrl: string, tip: string,
        obrisan: boolean, zemlja: string, takmicenje: string, status: string, timUrl: string, adminId: string){
       
             this.imeIgraca=imeIgraca;
             this.tim=tim;
             this.sezona=sezona;
             this.brend=brend;
             this.cena=cena;
             this.slikaUrl=slikaUrl;
             this.tip=tip;
             this.obrisan= obrisan;
             this.zemlja=zemlja;
             this.takmicenje=takmicenje;
             this.status=status;
             this.timUrl=timUrl;
             this.adminId=adminId
         }
  
       
 

}

export class DresUpdate {
    dresId:string
    imeIgraca: string
    tim: string
     sezona: string
     brend: string
     cena: number
     slikaUrl: string
     tip: string
     obrisan: boolean
     zemlja: string
     takmicenje: string
     status: string
     timUrl: string
   
    
     
     constructor(dresId:string, imeIgraca: string, tim:string, sezona: string, brend: string, cena: number, slikaUrl: string, tip: string,
        obrisan: boolean, zemlja: string, takmicenje: string, status: string, timUrl: string){
            this.dresId=dresId
            this.imeIgraca=imeIgraca;
            this.tim=tim;
            this.sezona=sezona;
            this.brend=brend;
            this.cena=cena;
            this.slikaUrl=slikaUrl;
            this.tip=tip;
            this.obrisan = obrisan;
            this.zemlja=zemlja;
            this.takmicenje=takmicenje;
            this.status=status;
            this.timUrl=timUrl;
           
        }
 
      


}
  
       
 
